### DISCUSSION TITLE
大佬，明明有些tag，使用get能从api到内容，但是bot机器人使用的post方式获取不到，怎么回事呢。

### DISCUSSION THREAD

**Discussion by Lixuhuilll** [Tsuk1ko/CQ-picfinder-robot#309] [问题求助]
就比如我在QQ群里发消息说来张老板涩图（当然，这是在整活），bot无法获取到图片，但是我使用get方式是能获取到内容的。
![image](https://user-images.githubusercontent.com/88485037/168003079-1cdf2397-212e-4ab4-8fa3-6d120f0fa08d.png)
![image](https://user-images.githubusercontent.com/88485037/168003118-8d10db89-571d-4fe5-a2ff-aa238d8af2e3.png)

**Comment by Tsuk1ko** [comment]
![image](https://user-images.githubusercontent.com/24877906/168005674-8e5dce4a-89e6-4e86-9823-a7d725842ce2.png)
因为参数连接符写错了，你实际传进去的参数根本不是这个

**Reply by Lixuhuilll** [reply]
那我这样应该没写错吧，api也返回了东西
![image](https://user-images.githubusercontent.com/88485037/168006011-9eff9935-b69f-43f3-a50d-948b5823e4da.png)

**Reply by Tsuk1ko** [reply]
我只能怀疑你用的请求工具有问题了，而且你看看你调用 api 的返回，里面 title author tags 里面有任何一个都不含有老板这个词啊……
![image](https://user-images.githubusercontent.com/24877906/168007034-91b40479-44ed-472f-8b49-7323b668c88f.png)

**Reply by Tsuk1ko** [reply]
老哥啊，你正文应该是 json 内容，content-type 是 header 啊你放正文里做什么

**Reply by Lixuhuilll** [reply]
好吧，我傻了。QAQ。
不过这个不能加个模糊搜索吗，感觉现在的搜索范围太窄了，经常报没图。

**Reply by Lixuhuilll** [reply]
或者我想添加更多的api，应该怎么做

**Reply by Lixuhuilll** [reply]
我没学过js，源码看得有点懵，只能看个大概。

**Reply by Tsuk1ko** [reply]
我在搜索上面已经在模糊和精确之间做了平衡了，没图很正常，这图都是我自己加的，我和你混的圈子又不是完全一样的，而且你要搜老板这种离谱的词那肯定是没图的，搜点动漫、二次元手游、动漫或游戏人物这种还是有图的，御姐萝莉这种属性分类当然也是有的

**Reply by Tsuk1ko** [reply]
> 或者我想添加更多的api，应该怎么做

这也没法说得清楚，只能告诉你文件在哪你看着改……

https://github.com/Tsuk1ko/cq-picsearcher-bot/blob/master/src/plugin/setu.js

**Reply by Lixuhuilll** [reply]
我找到了post的定义，这个D的结构在哪里啊，我找不到QAQ
  post<T = any, R = AxiosResponse<T>, D = any>(url: string, data?: D, config?: AxiosRequestConfig<D>): Promise<R>;

**Reply by Tsuk1ko** [reply]
data 就是 post 正文，如果你直接传对象就会自动按 json 给你 post
